tileserver/rgd.mjs

```diff
@@ -13,270 +11,271 @@ const logging = process.env.RDWATCH_VECTOR_TILE_LOGGING || false;
-      "rdwatch_modelrun"."title" AS "configuration_name",
-      "rdwatch_observationlabel"."slug" AS "label",
-      EXTRACT(
-        epoch
-        FROM
-          "rdwatch_siteevaluation"."timestamp"
-      ) :: bigint AS "timestamp",
-      EXTRACT(
-        epoch
-        FROM
-          "rdwatch_siteevaluation"."start_date"
-      ) :: bigint AS "timemin",
-      EXTRACT(
-        epoch
-        FROM
-          "rdwatch_siteevaluation"."end_date"
-      ) :: bigint AS "timemax",
-      "rdwatch_modelrun"."performer_id" AS "performer_id",
-      "rdwatch_performer"."slug" AS "performer_name",
-      "rdwatch_region"."name" AS "region",
-      CASE
-        WHEN (
-          "rdwatch_performer"."slug" = 'TE'
-          AND "rdwatch_siteevaluation"."score" = 1.0
-        ) THEN TRUE
-        ELSE FALSE
-      END AS "groundtruth",
-      "rdwatch_siteevaluation"."number" AS "site_number",
-      CASE
-        WHEN COUNT("rdwatch_siteobservation"."id") = 0 THEN TRUE
-        ELSE FALSE
-      END AS "site_polygon"
-    FROM
-      "rdwatch_siteevaluation"
-      INNER JOIN "rdwatch_modelrun" ON (
-        "rdwatch_siteevaluation"."configuration_id" = "rdwatch_modelrun"."id"
-      )
-      INNER JOIN "rdwatch_region" ON (
-        "rdwatch_siteevaluation"."region_id" = "rdwatch_region"."id"
-      )
-      INNER JOIN "rdwatch_observationlabel" ON (
-        "rdwatch_siteevaluation"."label_id" = "rdwatch_observationlabel"."id"
-      )
-      LEFT OUTER JOIN "rdwatch_siteobservation" ON (
-        "rdwatch_siteevaluation"."id" = "rdwatch_siteobservation"."siteeval_id"
-      )
-      INNER JOIN "rdwatch_performer" ON (
-        "rdwatch_modelrun"."performer_id" = "rdwatch_performer"."id"
-      )
-    WHERE
-      (
-        "rdwatch_siteevaluation"."configuration_id" = $4
-        AND ST_Intersects(
+function query(z, x, y, modelRunId, year) {
+  return sql`
+    WITH evaluations AS (
+      SELECT
+        "rdwatch_siteevaluation"."id",
+        "rdwatch_siteevaluation"."timestamp",
+        "rdwatch_siteevaluation"."configuration_id",
+        "rdwatch_siteevaluation"."number",
+        "rdwatch_siteevaluation"."start_date",
+        "rdwatch_siteevaluation"."end_date",
+        "rdwatch_siteevaluation"."geom" :: bytea,
+        "rdwatch_siteevaluation"."label_id",
+        "rdwatch_siteevaluation"."score",
+        "rdwatch_siteevaluation"."version",
+        "rdwatch_siteevaluation"."notes",
+        "rdwatch_siteevaluation"."validated",
+        "rdwatch_siteevaluation"."status",
+        "rdwatch_siteevaluation"."cache_originator_file",
+        "rdwatch_siteevaluation"."cache_timestamp",
+        "rdwatch_siteevaluation"."cache_commit_hash",
+        "rdwatch_siteevaluation"."id" AS "id",
+        "rdwatch_siteevaluation"."id" AS "uuid",
+        ST_AsMVTGeom(
           "rdwatch_siteevaluation"."geom",
-          ST_TileEnvelope($1, $2, $3)
+          ST_TileEnvelope(${z}, ${x}, ${y})
```

Comment:
This _looks like_ it's vulnerable to SQL injection, but `postgres.js` actually sanitizes these (see [docs](https://github.com/porsager/postgres#query-parameters)) via the `sql` template literal.

Reply:
nice, a lot better than continually looking up the params.